freshmaker/errata.py

```diff
@@ -392,25 +411,30 @@ def get_pulp_repository_ids(self, errata_id):
             '/errata/get_pulp_packages/{}.json'.format(errata_id))
         return data.keys()
 
-    def get_cve_affected_srpm_nvrs(self, errata_id):
-        """ Get SRPM nvrs which are affected by the CVEs in errata
+    def get_cve_affected_rpm_nvrs(self, errata_id):
+        """ Get RPM nvrs which are affected by the CVEs in errata
 
         :param errata_id: Errata advisory ID, e.g. 25713.
         :type errata_id: str or int
-        :return: a list of strings each of them is a source rpm nvr
+        :return: a list of strings each of them is a binary rpm nvr
         :rtype: list
         """
         data = self._errata_rest_get(f"/erratum/{errata_id}/builds_by_cve")
-        srpms = set()
-        for cve, data_by_product in data.items():
-            for product, product_data in data_by_product.items():
+        final_rpms = set()
+        for data_by_product in data.values():
+            for product_data in data_by_product.values():
                 for build in product_data.get("builds", []):
-                    for build_nvr, build_info in build.items():
+                    for build_info in build.values():
                         # for rpm advisories, build_nvr equal to srpm nvr, but this is
                         # not true for module advisories, so we need to get the srpms
                         # from variants data
                         for variant, variant_data in build_info.get("variant_arch", {}).items():
-                            for srpm in variant_data.get("SRPMS", []):
-                                srpms.add(srpm.split(".src.rpm")[0])
-
-        return list(srpms)
+                            for arch, rpms in variant_data.items():
+                                # Do we need to exclude 'noarch' too?
+                                # Remove '.arch.....' part from rpm's name
+                                # and make a list from them
+                                if arch != 'SRPMS':
+                                    just_nvrs = [rpm.partition("." + arch)[0] for rpm in rpms]
```

Comment:
You can use `rpm.rsplit('.', 2)[0]`, not sure if there could be some cases that arch doesn't match with the string in rpm.

Reply:
I don't know, I think the arch and the part of string should be the same. And I am not sure that it's safe to just split by dot, because I think there could be different amount of dots at the end of string.

Reply:
RPM filenames always end with .<arch>.rpm. There are cases where the arch in the rpm can vary (i386, i586, i686). The `rsplit()` method is more reliable.

Reply:
Ok, I've changed it to `rsplit()`